src/lib.rs

```diff
@@ -26,75 +26,107 @@
 #![deny(unused_imports)]
 #![deny(missing_docs)]
 
+#![cfg_attr(all(not(test), not(feature = "std")), no_std)]
+#[cfg(any(test, feature = "std"))] pub extern crate core;
+
 extern crate bitcoin_hashes;
+#[cfg(feature = "std")]
 extern crate unicode_normalization;
+
 #[cfg(feature = "rand")]
 extern crate rand;
+#[cfg(feature = "serde")]
+pub extern crate serde;
 
+#[cfg(feature = "std")]
 use std::{error, fmt, str};
+#[cfg(feature = "std")]
 use std::borrow::Cow;
 
 use bitcoin_hashes::{sha256, Hash};
+
+#[cfg(feature = "std")]
 use unicode_normalization::UnicodeNormalization;
 
+#[macro_use]
+mod internal_macros;
 mod language;
 mod pbkdf2;
 
 pub use language::Language;
 
-/// The ideagrapic space that should be used for Japanese lists.
-#[cfg(feature = "japanese")]
+/// The minimum number of words in a mnemonic.
 #[allow(unused)]
-const IDEOGRAPHIC_SPACE: char = '　';
+const MIN_NB_WORDS: usize = 12;
+
+/// The maximum number of words in a mnemonic.
+const MAX_NB_WORDS: usize = 24;
+
+/// A structured used in the [Error::AmbiguousLanguages] variant that iterates
+/// over the possible languages.
+#[derive(Debug, Clone, PartialEq, Eq, Copy)]
+pub struct AmbiguousLanguages([bool; language::MAX_NB_LANGUAGES]);
+
+impl AmbiguousLanguages {
+	/// Presents the possible languages in the form of a slice of booleans
+	/// that correspond to the occurrences in [Language::all()].
+	pub fn as_bools(&self) -> &[bool; language::MAX_NB_LANGUAGES] {
+		&self.0
+	}
+
+	/// An iterator over the possible languages.
+	#[cfg(feature = "std")]
+	pub fn iter(&self) -> impl Iterator<Item = Language> + '_ {
+		Language::all().iter().enumerate().filter(move |(i, _)| self.0[*i]).map(|(_, l)| *l)
```

Comment:
Oh uhm, I'm not aware of `core`, really. What's the exact difference? I think many uses of `std` can be changed for `core` here. But that's not a breaking change, so we can do that later?

Reply:
`core` is always available afaik, even without `std` or `alloc`. Iirc it goes like this: `core` ⊂ `std` and `alloc` ⊂ `std`. I think this crate only needs `core` ∪ `alloc` (doesn't do any fs/net io or other weird things). So it can always be no-std, and the things currently marked as `std` only really need `alloc`. But yes, that can be done as another release (some stuff as 1.1, other stuff probably as 2.0).